application/src/test/java/be/jpendel/domain/event/EventFactoryTest.java

```diff
@@ -0,0 +1,24 @@
+package be.jpendel.domain.event;
+
+import be.jpendel.application.CreateEventCommand;
+import org.junit.Test;
+
+import java.time.LocalDateTime;
+
+public class EventFactoryTest {
+
+    @Test
+    public void eventWithSameNameAndSameStartdateLocationCanNotBeCreated(){
+       EventFactory factory = new EventFactory();
+        LocalDateTime time = LocalDateTime.now();
+        CreateEventCommand eventCmd = CreateEventCommand.builder()
+                .withEndDateTime(time).withLocation("loc")
+                .withStartDateTime(time)
+                .withName("test").build();
+        Event e1 = factory.create(eventCmd);
+        EventRepository repo = new InMemEventRepository();
+        repo.add(e1);
+        repo.add(e1);// moet dit falen ?
```

Comment:
Ik zou het zeker ook in de repo laten falen. Je kan in je applicatie service, of in de factory de check doen of er al een event met dezelfde naam en zo bestaat. Maar op moment van persisteren moet het zeker gevalideerd worden.